src/libxtp/ERIs.cc

```diff
@@ -192,7 +192,116 @@ namespace votca {
         }
         
         
-        
+		
+		
+		void ERIs::CalculateERIs_4c_direct(const AOBasis& dftbasis, const ub::matrix<double> &DMAT)
+		{
+			cout << endl;
+			cout << "ERIS.cc ERIs::CalculateERIs_4c_direct" << endl;
+			
+			// Initialize ERIs matrix
+			_ERIs = ub::zero_matrix<double>(DMAT.size1(), DMAT.size2());
+			// Get number of shells
+			int numShells = dftbasis.getNumofShells();
+			
+			// Test
+			printf("size(DMAT) = [%d, %d]\n", int(DMAT.size1()), int(DMAT.size2()));
+			printf("dftbasisSize = %d\n", dftbasis.AOBasisSize());
+			printf("numShells = %d\n", numShells);
+
+			// Timer
+			clock_t start = clock();
+
+			#pragma omp parallel for
+			for (int iShell_3 = 0; iShell_3 < numShells; iShell_3++)
+			{
+				const AOShell* shell_3 = dftbasis.getShell(iShell_3); // Shell object
```

Comment:
There are several shared objects here have you run any calculations to see how efficient the parallelization is. If you have shared memory access between threads the performance can actually be worse than running it in serial.

Reply:
Yes and no, 1) The shell objects are pretty large, so it is unlikely that false sharing happens. Secondly, the are read access only, which also reduces the risk. I agree that maybe we should rewrite AOBasis as a continguous vector without any pointers.